src/Components/Card/ProductCard.js

```diff
@@ -17,12 +25,121 @@ export default function ProductCard(props)
+            setWishlistBtn("add-to-wishlist-btn")
+        }
+    },[userWishlist, productdetails._id, setWishlistHeartIcon, setWishlistBtn])
+
+    async function addOrRemoveItemToWishlist()
+    {
+        if(wishlistHeartIcon==="fa-heart-o" && wishlistBtn ==="add-to-wishlist-btn")
+        {
+            //Item not present in wishlist, add it
+            const token=localStorage.getItem('token')
+
+            if(token)
+            {
+                const user = jwt_decode(token)
+                
+                if(!user)
+                {
+                    localStorage.removeItem('token')
+                    showToast("warning","","Kindly Login")
+                    navigate('/login')
+                }
+                else
+                {
+                    let wishlistUpdateResponse = await axios.patch(
+                        "https://bookztron.herokuapp.com/api/wishlist",
+                        {
+                            productdetails
+                        },
+                        {
+                            headers:
+                            {
+                                'x-access-token': localStorage.getItem('token'),
+                            }
+                        }
+                    )
+            
+                    if(wishlistUpdateResponse.data.status==="ok")
+                    {
+                        setWishlistHeartIcon("fa-heart")
+                        setWishlistBtn("added-to-wishlist-btn")
+                        dispatchUserWishlist({type: "UPDATE_USER_WISHLIST",payload: wishlistUpdateResponse.data.user.wishlist})
+                        showToast("success","","Item successfully added to wishlist")
+                    }
+                }
+            }
+            else
+            {
+                showToast("warning","","Kindly Login")
+            }   
+        }
+        else
+        {
+            //Item present in wishlist, remove it
+            const token=localStorage.getItem('token')
+
+            if(token)
+            {
+                const user = jwt_decode(token)
+                
+                if(!user)
+                {
+                    localStorage.removeItem('token')
+                    showToast("warning","","Kindly Login")
+                    navigate('/login')
+                }
+                else
+                {
+                    let wishlistUpdateResponse = await axios.delete(
+                        `https://bookztron.herokuapp.com/api/wishlist/${productdetails._id}`,
+                        {
+                            headers:
+                            {
+                                'x-access-token': localStorage.getItem('token'),
+                            }
+                        },
+                        {
+                            productdetails
+                        }
+                    )
+                    if(wishlistUpdateResponse.data.status==="ok")
```

Comment:
Suggestion: you should use `200` or `201` for checking response status, because The `ok` contains a Boolean stating whether the response was successful (status in the range 200-299) or not. Correct me if I am wrong.